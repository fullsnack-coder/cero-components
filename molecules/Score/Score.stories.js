import { Score, styles } from '.'

import { getTemplate } from '@glrodasz/storybook-tools-helpers'

const Template = getTemplate(Score, styles)

export default {
  title: 'Molecules/Score',
  component: Score,
  args: {},
}

export const Default = Template.bind({})
