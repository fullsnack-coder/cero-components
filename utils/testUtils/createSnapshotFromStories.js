import React from 'react'
import { render } from '@testing-library/react'
import { composeStories } from '@storybook/testing-react'

// TODO: Move this to a generic approach where it
// reads all the stories and works as Storyshoot
const createSnapshotFromStories = (componentName, stories) => {
  const composedStories = composeStories(stories)

  Object.entries(composedStories).forEach(([story, Component]) => {
    describe(`when '${componentName}' ${story} is mounted`, () => {
      it('should render', () => {
        const { asFragment } = render(<Component />)
        expect(asFragment()).toMatchSnapshot()
      })
    })
  })
}

export default createSnapshotFromStories
