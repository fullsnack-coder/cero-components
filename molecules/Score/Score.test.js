import * as stories from './Score.stories'
import createSnapshotFromStories from '../../utils/testUtils/createSnapshotFromStories'

describe('[ molecules / Score ]', () => {
  describe('#Score', () => createSnapshotFromStories('Score', stories))
})
