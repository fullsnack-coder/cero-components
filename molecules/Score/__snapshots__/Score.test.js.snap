// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ molecules / Score ] #Score when 'Score' Default is mounted should render 1`] = `
<DocumentFragment>
  <div
    class="score"
  >
    <div
      class="face"
    >
      <svgrurl />
      <div
        class="spacer vertical"
        style="height: 5px;"
      />
      <p
        class="paragraph color-base size-md weight-normal"
      >
        Frustrado
      </p>
    </div>
    <div
      class="spacer horizontal"
      style="height: 100%; max-height: auto; width: 40px;"
    />
    <div
      class="face"
    >
      <svgrurl />
      <div
        class="spacer vertical"
        style="height: 5px;"
      />
      <p
        class="paragraph color-base size-md weight-normal"
      >
        Normal
      </p>
    </div>
    <div
      class="spacer horizontal"
      style="height: 100%; max-height: auto; width: 40px;"
    />
    <div
      class="face"
    >
      <svgrurl />
      <div
        class="spacer vertical"
        style="height: 5px;"
      />
      <p
        class="paragraph color-base size-md weight-normal"
      >
        Muy motivado
      </p>
    </div>
  </div>
</DocumentFragment>
`;
